test/extensions/filters/http/common/fuzz/uber_filter.h

```diff
@@ -91,7 +91,7 @@ class UberFilterFuzzer {
       }
       Buffer::OwnedImpl buffer(data.data().Get(i));
       ENVOY_LOG_MISC(debug, "Decoding data: {} ", buffer.toString());
-      if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
+      if (filter->decodeData(buffer, end_stream) == Http::FilterDataStatus::StopIterationNoBuffer) {
```

Comment:
Can you explain a bit more what is going on here that requires that `StopIterationNoBuffer` is treated specially vs. the other `StopIteration*`?

Reply:
It looks like the HCM can continue to send data to decode in the filters for other StopIterations, while `StopIterationNoBuffer` would. While I think all the decode paths can still be hit with the right combination of data/end_stream values in a stateless scenario like this, I think @nareddyt needed this to hit some code in grpc transcoding filter. @nareddyt  what was the particular state that came up for grpc transcoding?

Reply:
I don't think that's the case. When you stop iteration, you don't continue the filter chain.

Reply:
Hmm my read on it when we were discussing was the `StopIterationAnd*` followed by a continueDecoding() would make HCM resume calling decodeData, but if this actually means it continues to subsequent filters, then the original was correct

Reply:
If a filter returns `StopIterationAndBuffer`, then more data will be continue being sent to that filter (correct?).

With the pre-existing code, a `StopIterationAndBuffer` results in no more `decodeData` calls to that filter (before `end_stream` is true).

Reply:
@nareddyt that's correct, but that also applies to the other `StopIteration*`, i.e. they can be invoked again (just not any filter down the chain). Watermark has a contract that is a bit different, but it's totally reasonable to have a buffer with `StopIterationAndBuffer` or `StopIterationNoBuffer` be in the business of accumulating incoming data for transcoding.

Reply:
I think it comes down to: will continuing to call decode* exploring more codepaths than individual iterations? Unless a filter changes some internal state before returning StopIteration*, and subsequent decode*'s will be different than decode*'s before returning StopIteration*, does this change make the target more likely to hit a bug in the filter?

Reply:
Sure, to the extent we can make more of this individual steps under the control of LPM and the fuzzer, the better. Maybe we should lift all this logic out and never do a continue?

Reply:
Like a single headers run, single data run, single trailer run?

Reply:
Thinking about it a bit, looking at your previous comment, I think my answer is "don't know; depends on the filter". Do you plan on having other potential actions around fake RPC servers or things that the filter might be reaching out to as step actions? Would that then allow filters to trigger a continuation?

One intriguing possibility is what we were discussing earlier around having filters include a declarative manifest of the environment they interact with, e.g. which RPC services, which filters providing data dependencies. If we had this, we could build this fake environment out.